app/controllers/tasks_controller.rb

```diff
@@ -4,42 +4,72 @@ def index
     render :index
   end
 
+  # Display a specific task
   def show
-    @task = Task.find(params[:id])
+    @task   = Task.find(params[:id])
     render :tasks
   end
 
   def new
     @task = Task.new
-
     render :new
   end
 
-  # Add and save a new task to db
+  # Add and save a new task to list
   def create
-    @task = Task.new(task_params[:task])
-    @task.save
-
+    task = Task.new(task_params[:task])
+    task.save
     index
   end
 
-  # Delete a task from db
+  # Delete a task from list
   def destroy
-    @task = Task.find(params[:id])
-    @task.destroy.save
-
+    task = Task.find(params[:id])
+    task.destroy.save
     redirect_to '/'
   end
 
+  # Update an existing task
+  def edit
+    @task = Task.find(params[:id])
+    render :update
+  end
+
   def update
     @task = Task.find(params[:id])
-    @task.update.save
+    @name = task_params[:task][:name]
+    @desc = task_params[:task][:desc]
+    @comp_date = task_params[:task][:comp_date]
+    @person_id = task_params[:task][:person_id]
+
+    @task.update(name: @name)
+    @task.update(desc: @desc)
+    @task.update(comp_date: @comp_date)
+    @task.update(person_id: @person_id)
```

Comment:
I didn't realize that you could do this at the time. This cleans things up so much! Thanks.